### PR TITLE
use lookaside storage for remote tests

### PR DESCRIPTION
**Pull request by baude**
in an effort to speed up the remote testing, we should be using
lookaside storage to avoid pull images as well as importing multiple
images into the RW store.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
/me looks for `int remote`, sees "Successful in 35-49m", weeps tears of joy

**Comment by baude** [comment]
ready for review.

File: test/e2e/images_test.go [hunk 1]
```diff
@@ -45,18 +45,6 @@ var _ = Describe("Podman images", func() {
 		Expect(session.LineInOuputStartsWith("quay.io/libpod/busybox")).To(BeTrue())
 	})
 
-	It("podman images with no images prints header", func() {
-		rmi := podmanTest.PodmanNoCache([]string{"rmi", "-a"})
```
Comment (rhatdan) [inline test/e2e/images_test.go:49]: This means we can no longer test this functionality?  This seems bad?

File: test/e2e/images_test.go [hunk 2]
```diff
@@ -199,33 +191,23 @@ WORKDIR /test
...
-		podmanTest.RestoreAllArtifacts()
-		rmi := podmanTest.PodmanNoCache([]string{"image", "rm", "busybox"})
-		rmi.WaitWithDefaultTimeout()
-		Expect(rmi).Should(Exit(0))
-
 		dockerfile := `FROM quay.io/libpod/alpine:latest
 `
 		podmanTest.BuildImage(dockerfile, "foobar.com/before:latest", "false")
-		result := podmanTest.PodmanNoCache([]string{"image", "list", "-q", "-f", "after=quay.io/libpod/alpine:latest"})
+		result := podmanTest.Podman([]string{"image", "list", "-q", "-f", "after=quay.io/libpod/alpine:latest"})
```
Comment (rhatdan) [inline test/e2e/images_test.go:207]: This test is no longer checking after.

We need to add an image, and then test after it and make sure we don't get any images.

File: test/e2e/images_test.go [hunk 3]
```diff
@@ -341,22 +323,21 @@ WORKDIR /test
...
 RUN touch test.txt
 ENV foo=bar
 `
 		podmanTest.BuildImage(dockerfile, "test", "true")
-		session := podmanTest.PodmanNoCache([]string{"images"})
+		session := podmanTest.Podman([]string{"images"})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
-		Expect(len(session.OutputToStringArray())).To(Equal(5))
+		Expect(len(session.OutputToStringArray())).To(Equal(len(CACHE_IMAGES) + 2))
```
Comment (rhatdan) [inline test/e2e/images_test.go:335]: I like this.

File: test/e2e/import_test.go [hunk 4]
```diff
@@ -62,11 +62,9 @@ var _ = Describe("Podman import", func() {
 		export.WaitWithDefaultTimeout()
 		Expect(export.ExitCode()).To(Equal(0))
 
-		importImage := podmanTest.PodmanNoCache([]string{"import", outfile})
+		importImage := podmanTest.Podman([]string{"import", outfile})
 		importImage.WaitWithDefaultTimeout()
 		Expect(importImage.ExitCode()).To(Equal(0))
-
-		Expect(podmanTest.ImageExistsInMainStore(importImage.OutputToString())).To(BeTrue())
```
Comment (rhatdan) [inline test/e2e/import_test.go:69]: podman images shows readwrite and readonly images, we could get these from this command.

File: test/e2e/libpod_suite_test.go [hunk 5]
```diff
@@ -50,7 +37,7 @@ func (p *PodmanTestIntegration) setDefaultRegistriesConfigEnv() {
 func (p *PodmanTestIntegration) setRegistriesConfigEnv(b []byte) {
 	outfile := filepath.Join(p.TempDir, "registries.conf")
 	os.Setenv("REGISTRIES_CONFIG_PATH", outfile)
-	ioutil.WriteFile(outfile, b, 0644)
+	ioutil.WriteFile(outfile, b, 0654)
```
Comment (rhatdan) [inline test/e2e/libpod_suite_test.go:40]: What is this change about?

**Review by rhatdan — COMMENTED** [review]
My fear with this is you gone too far and some features of podman not longer get exercised.

podman rmi -a
podman rmi -af

Are we actually still pulling any images from registries?  
podman pull
podman run
podman build
podman create

File: test/e2e/images_test.go (hunk 1 above)
Comment (baude) [reply test/e2e/images_test.go:49]: it got moved to a system test and is in Ed's pr

File: test/e2e/libpod_suite_test.go (hunk 5 above)
Comment (baude) [reply test/e2e/libpod_suite_test.go:40]: i have no idea!  i didnt intentionally change it.

File: test/e2e/images_test.go [hunk 6]
```diff
@@ -131,45 +119,49 @@ var _ = Describe("Podman images", func() {
...
 		Expect(session).Should(Exit(0))
 		Expect(len(session.OutputToStringArray())).To(Equal(2))
 	})
 
 	It("podman images filter reference", func() {
-		podmanTest.RestoreAllArtifacts()
-		result := podmanTest.PodmanNoCache([]string{"images", "-q", "-f", "reference=quay.io*"})
+		waht := podmanTest.Podman([]string{"images"})
+		waht.WaitWithDefaultTimeout()
+		fmt.Println(waht.OutputToString())
```
Comment (TomSweeneyRedHat) [inline test/e2e/images_test.go:146]: Doesn't really matter, but "waht"?  Just curious.

Comment (baude) [reply test/e2e/images_test.go:146]: haha

**Comment by TomSweeneyRedHat** [comment]
LGTM other than other comments

**Comment by baude** [comment]
> My fear with this is you gone too far and some features of podman not longer get exercised.
> 
> podman rmi -a
> podman rmi -af
> 
> Are we actually still pulling any images from registries?
> podman pull
> podman run
> podman build
> podman create

Yes, we are still pulling images.  Looks for cirros. Also, podman rmi -a is exercised in several tests still.

**Pull request merged**